src/Nlog.RabbitMQ.Target/RabbitMQTarget.cs

```diff
@@ -334,35 +334,45 @@ private byte[] GetMessage(LogEventInfo logEvent)
 		}
 
 		private string GetMessageString(LogEventInfo logEvent)
-		{
-			if (!UseJSON)
-			{
-				return RenderLogEvent(Layout, logEvent);
-			}
-			else
-			{
-				var message = this.UseLayoutAsMessage ? RenderLogEvent(Layout, logEvent) : logEvent.FormattedMessage;
-				var messageSource = RenderLogEvent(MessageSource, logEvent);
-				if (string.IsNullOrEmpty(messageSource))
-					messageSource = string.Format("nlog://{0}/{1}", System.Net.Dns.GetHostName(), logEvent.LoggerName);
-
-				try
-				{
-					var jsonSerializer = JsonSerializer;
-					lock (jsonSerializer)
-					{
-						return MessageFormatter.GetMessageInner(jsonSerializer, message, messageSource, logEvent, this.Fields);
-					}
-				}
-				catch (Exception e)
-				{
-					_jsonSerializer = null;	// reset as it it might be broken
-					InternalLogger.Error(e, "RabbitMQTarget(Name={0}): Failed to serialize LogEvent: {1}", Name, e.Message);
-					throw;
-				}
-			}
+		{
+			if (!UseJSON)
+			{
+				return RenderLogEvent(Layout, logEvent);
+			}
+			else
+			{
+				var message = this.UseLayoutAsMessage ? RenderLogEvent(Layout, logEvent) : logEvent.FormattedMessage;
+				var messageSource = RenderLogEvent(MessageSource, logEvent);
+				if (string.IsNullOrEmpty(messageSource))
+					messageSource = string.Format("nlog://{0}/{1}", System.Net.Dns.GetHostName(), logEvent.LoggerName);
+
+				try
+				{
+					var jsonSerializer = JsonSerializer;
+					lock (jsonSerializer)
+                    {
+						return MessageFormatter.GetMessageInner(jsonSerializer, message, messageSource, logEvent, Fields, GetFullContextProperties(logEvent));
+					}
+				}
+				catch (Exception e)
+				{
+					_jsonSerializer = null;	// reset as it it might be broken
+					InternalLogger.Error(e, "RabbitMQTarget(Name={0}): Failed to serialize LogEvent: {1}", Name, e.Message);
+					throw;
+				}
+			}
 		}
 
+        private IDictionary<string, object> GetFullContextProperties(LogEventInfo logEvent)
+        {
+            var allProperties = GetContextProperties(logEvent) ?? new Dictionary<string, object>();
+            var ndlcProperties = GetContextNdlc(logEvent);
+            if (IncludeNdlc && ndlcProperties.Count > 0)
+                allProperties.Add("nested", ndlcProperties);
```

Comment:
Could the code become like this:

```c#
if (IncludeNldc)
{
        var nldcProperties = GetContextNdlc(logEvent);
        if (nldcProperties.Count > 0)
              allProperties.Add("nested", ndlcProperties);
}
```

Maybe consider changing the name from "nested" to "scope". And make a new target property like this:

```c#
public bool IncludeScope { get => IncludeNdlc; set => IncludeNdlc = value; }
```

Reply:
I don't understand why you'd like to add a new property `IncludeScope ` while the original (included in `TargetWithContext` ) is exactly what it is.

Reply:
Because I think `allProperties.Add("scope", ndlcProperties);` is a more meaningful property-name.

Reply:
But if you don't like to create this new artificial property. Then I think it should be `allProperties.Add("ndlc", ndlcProperties);`. But that propery-name only makes sense for die-hard NLog-users.

Reply:
I would prefer use `allProperties.Add("ndlc", ndlcProperties);` rather than `allProperties.Add("scope", ndlcProperties);`, because if you use `ILogger `from `microsoft.extensions.logging` and you do `_logger.BeginScope(...)` (https://github.com/NLog/NLog.Extensions.Logging/wiki/NLog-properties-with-Microsoft-Extension-Logging) it will use mdlc and not ndlc. Then it will be very confusing. 
I thought about "nested" first because nldc is shortcut for "Nested Diagnostics Logical Context ", but NLog use ndlc naming and "IncludeNdlc", so I better use the same naming.

Reply:
Actually BeginScope will inject into both NDLC and MDLC. Which is why `${ndlctiming}` is working with BeginScope.

Reply:
But mdlc is already contained in GetAllProperties, only ndlc is missing. Do you still prefer scope while there is only ndlc values?

Reply:
I'm happy if the property is called "ndlc", but if you change the property-name, then I think you should create the artificial property to match the property-name to make it clear for users. But I was just thinking that "ndlc" might not be the most obvious property-name for users outside the NLog-core-community. (Btw. please change the logic, so you perform the `IncludeNldc`-check first as shown in my original review-comment)

Reply:
Well If you use nldc, you'll surely know what it is. It's the name in the NLog documentationy I cannot find a better name without changing it in NLog as well.